# ![DRAGON logo](https://github.com/qiyf/images/blob/master/logo2.png)

DRAGON is a software package to enable De novo, and RAtional prediction of Genome organizatiON. It provides an implementation of the model proposed in the [manuscript](https://www.biorxiv.org/content/early/2018/03/15/282095) to simulate chromatin structure and dynamics. With DRAGON, one can predict the structure of a 25Mb long chromatin region from a variety of cell types using genome-wide profiles of histone modifications and CTCF molecules. 

The package is mainly written in Python, and it streamlines all the necessary steps to process [epigenomics data](./processEpigenomicsData/), to perform [molecular dynamics simulations](./runMolecularDynamics/) and to analyze [predicted conformational ensemble](./analyzeChromatinConformation/) for the chromatin. 

## Installation
DRAGON can be downloaded and installed by running the following command:
```
git clone https://github.com/ZhangGroup-MITChemistry/DRAGON.git
```
or download the zip file with the link:

[https://github.com/ZhangGroup-MITChemistry/DRAGON/archive/master.zip](https://github.com/ZhangGroup-MITChemistry/DRAGON/archive/master.zip)  

After the installation of DRAGON, one will need to install and compile [LAMMPS package](http://lammps.sandia.gov/) to enable molecular dynamics simulations. This can be done by executing the following command:

```
./src/LAMMPS.sh
```

Note that [GCC](https://gcc.gnu.org/) compiler module need to be installed beforehand and an environment of [OpenMPI](https://www.open-mpi.org/) is needed to compile the parallel version of LAMMPS. 

Note that a FORTRAN compiler (e.g. ifort or gfortran) also need to be available for calculating contact map step. Refer to **Calculate and visualize contact map** in section **III) Analyze Chromatin Conformation** in the following and [README](./analyzeChromatinConformation/contactMap/README.md) for more detailed information. 

## Usage

We model the chromatin as beads on a string. Each bead is assigned with a chromatin state, and will be labeled as a CTCF binding site in a given orientation depending on the underlying Chip-Seq signal. 

The exact process is illustrated with the following flow chart. 

![Flow chart](https://github.com/qiyf/images/blob/master/flow_chart.png)

Follow the following steps to initialize a chromatin structure simulation of chromosome 1 from GM12878 cell line. All the executable scripts are provided in the [`./example/`](./example/) folder. 

### I) Process Epigenomics Data

```
./example/1-processEpigenomicsData.sh
```

This script displays the necessary components for epigenomics data processing. [ChromHMM](http://compbio.mit.edu/ChromHMM/) is used to process epigenomics data of genome-wide histone modifications from six cell types and define the chromatin states. See [README](./processEpigenomicsData/README.md) for details on its installation and usage. ChIP-Seq narrow peak signal for the CTCF and Cohesin are used to define the CTCF-binding sites. 

### II) Run Molecular Dynamics Simulation
We use LAMMPS to simulate chromatin structure and dynamics. See [README](./runMolecularDynamics/README.md) for its detailed usage. For the steps in this section, a [main python program](./runMolecularDynamics/main.py) is provided to incorporate all necessary steps and initialize simulations with different cell types and multiple chromosomes. See [README](./runMolecularDynamics/README.md) for detailed usage of that program.

#### Select a 25Mb chromatin region
```
./example/2-selectChromatinRegion.sh
```

A [chromtin region file](./src/chr_region.txt) that indicates the 25Mb long chromatin region that is of interested for each chromosome is generated by executeing this script. The format is in the following:
```
chromosome_id     start_position(Mb)      end_position(Mb)  
1                 20                      45  
2                 20                      45  
3                 20                      45  
4                 20                      45   
```

If a different 25Mb chromatin region for any individual chromosome is desired, change the start and end positions either in the original script [`./example/2-selectChromatinRegion.sh`](./example/2-selectChromatinRegion.sh) and regenerate the chromatin regions or simply changed the chromatin positions in the generated [chromtin region file](./src/chr_region.txt) and continue the later steps.

#### Extract epigenomics input

```
./example/3-extractEpigenomicsInput.sh
```

Files for Chromatin states and CTCF-binding sites as the model input for the specific 25Mb chromatin region indicated in  [chromtin region file](./src/chr_region.txt) are generated by executing this script: 20Mb to 45Mb for chromosome 1, GM12878 under this case. The input files are located at [`./runMolecularDynamics/inputFiles/epig_input/`](./runMolecularDynamics/inputFiles/epig_input/). See [Chromatin States README](./runMolecularDynamics/inputFiles/epig_input/chromStates/README.md) and [CTCF-binding Sites README](./runMolecularDynamics/inputFiles/epig_input/ctcfSites/README.md) for details on processing the data extraction.

#### Build LAMMPS input

```
./example/4-buildLammpsInput.sh
```

LAMMPS input files are generated by executing this script. The input files are located at [`./runMolecularDynamics/inputFiles/lmps_input/`](./runMolecularDynamics/inputFiles/lmps_input/) and in the simulation folder [`./runMolecularDynamics/run_folder/Gm12878/chr1/run00/`](./runMolecularDynamics/run_folder/Gm12878/chr1/run00/). 

#### Run simulation

```
./example/5-runMD.sh
```

A single simulation for chromosome 1, GM12878 is started by executing this script. The trajectory files are dumped as .dcd file located in the simulation folder [`./runMolecularDynamics/run_folder/Gm12878/chr1/run00/`](./runMolecularDynamics/run_folder/Gm12878/chr1/run00/). Note that this is to run the simulation in serial, which might be relatively slow but simple enough to be represented as part of the instruction. 

### III) Analyze Chromatin Conformation

We use [MATLAB](https://www.mathworks.com/products/matlab.html) to analyze contact maps and [VMD](http://www.ks.uiuc.edu/Research/vmd/) to visualize chromatin structure. Installation of these two software packages is highly recommended. See [contact map README](./analyzeChromatinConformation/contactMap/README.md) and [structure visualization README](./analyzeChromatinConformation/visStructure/README.md) for detailed instructions on usage. 

#### Calculate and visualize contact map

```
./example/6-calcCMAP.sh
```

The simulated contact map for chromosome 1, GM12878 is calculated by executing this script. The core program to calculate the contact map from trajectory files is a FORTRAN code and has been compiled located at [`./src/cmap/FORTRAN/`](./src/cmap/FORTRAN/) with ifort compiler. It can also be compiled with gfortran, but have either of these compilers installed beforehand is necessary. 

The calculated contact map is located at [`./analyzeChromatinConformation/contactMap/cmap/`](./analyzeChromatinConformation/contactMap/cmap/). To visualize the contact map, use the provided [MATLAB script](./analyzeChromatinConformation/contactMap/visContactMap.m). See [README](./analyzeChromatinConformation/contactMap/README.md) for more detailed instructions. 

#### Visualize the 3D structure

```
./example/7-geneVMDScript.sh
```

The VMD script for the visualization of 3D structure for chromosome 1, GM12878 is generated by executing this script. The generated script is located at [`./analyzeChromatinConformation/visStructure/`](./analyzeChromatinConformation/visStructure/).  See [README](./analyzeChromatinConformation/visStructure/README.md) for detailed instruction of visualization steps with VMD.
